src/main/haskell/kore/src/Kore/Step/Function/Registry.hs

```diff
@@ -87,9 +93,34 @@ axiomToIdAxiomPatternPair
 axiomPatternsToEvaluators
     :: Map.Map (Id level) [AxiomPattern level]
     -> Map.Map (Id level) [ApplicationFunctionEvaluator level]
-axiomPatternsToEvaluators = fmap (map axiomPatternToEvaluator)
+axiomPatternsToEvaluators axiomPatterns =
+    -- remove the key if there are no associated function evaluators
+    Map.filter (not . null)
+        (mapMaybe axiomPatternEvaluator <$> axiomPatterns)
```

Comment:
So, if I understand correctly, currently none of the generated axioms will be added as evaluators due to the filtering below, as only assoc, comm, and unit were of the new form we accept (the others like functional or subsort have an existential quantifier so they don't qualify).  So we'll only have the hand-written axioms in the prelude for now. Is that right?

Reply:
The generated axioms for `append` are added. Axioms marked `assoc`, `comm`, and `unit` are filtered out below because we don't implement matching for general equalities. The `inj` axioms are also filtered out because `inj` is a parametric symbol.